src/Libraries/Microsoft.Extensions.AI.OpenAI/OpenAIClientExtensions.cs

```diff
@@ -214,6 +214,20 @@ internal static FunctionCallContent ParseCallContent(BinaryData utf8json, string
         FunctionCallContent.CreateFromParsedArguments(utf8json, callId, name,
             static utf8json => JsonSerializer.Deserialize(utf8json, OpenAIJsonContext.Default.IDictionaryStringObject)!);
 
+    /// <summary>Gets a media type for an image based on the file extension in the provided URI.</summary>
+    internal static string ImageUriToMediaType(Uri uri)
+    {
+        string absoluteUri = uri.AbsoluteUri;
+        return
+            absoluteUri.EndsWith(".png", StringComparison.OrdinalIgnoreCase) ? "image/png" :
+            absoluteUri.EndsWith(".jpg", StringComparison.OrdinalIgnoreCase) ? "image/jpeg" :
+            absoluteUri.EndsWith(".jpeg", StringComparison.OrdinalIgnoreCase) ? "image/jpeg" :
+            absoluteUri.EndsWith(".gif", StringComparison.OrdinalIgnoreCase) ? "image/gif" :
+            absoluteUri.EndsWith(".bmp", StringComparison.OrdinalIgnoreCase) ? "image/bmp" :
+            absoluteUri.EndsWith(".webp", StringComparison.OrdinalIgnoreCase) ? "image/webp" :
+            "image/*";
+    }
+
```

Comment:
Do we really need to guess the mediaType ?

I wonder if the `Api` cannot resolve this shouldn't be the responsibility of the API. What happens in case of a `malicious.gif` file is sent as a `image/gif`, do we assume the responsibility? Wondering the security/treat modeling for this.

Reply:
> Do we really need to guess the mediaType ?

From where else would we get it? We may as well not add it if we're not going to assume it from the file extension.

None of this is trusted. If the service were to send a media-type, it could just as easily be as wrong as the file extension.

Reply:
@stephentoub Is just something I would like to understand the implications from the perspective of treat modeling/responsibilities.

We are assuming in the name of the caller, for something we can't know for sure, that was my point trying to enlighten this uncertainty.